cli/azd/pkg/azdo/project.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/core"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/operations"
+)
+
+// returns a process template (basic, agile etc) used in the new project creation flow
+func getProcessTemplateId(ctx context.Context, client core.Client) (string, error) {
+	processArgs := core.GetProcessesArgs{}
+	processes, err := client.GetProcesses(ctx, processArgs)
+	if err != nil {
+		return "", err
+	}
+	process := (*processes)[0]
+	return process.Id.String(), nil
+}
+
+// creates a new Azure Devops project
+func createProject(ctx context.Context, connection *azuredevops.Connection, name string, description string, console input.Console) (*core.TeamProjectReference, error) {
+	coreClient, err := core.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+
+	processTemplateId, err := getProcessTemplateId(ctx, coreClient)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching process template id %w", err)
+	}
+
+	capabilities := map[string]map[string]string{
+		"versioncontrol": {
+			"sourceControlType": "git",
+		},
+		"processTemplate": {
+			"templateTypeId": processTemplateId,
+		},
```

Comment:
One is allower, the other is camel...consistent?

Reply:
We can make the adjustment, unfortunately this is an underlying api property. In a future PR we could create a struct with the json mapping comments instead of relying on the map object. This could be a contribution back to the go sdk for azure devops and the capabilities property.